packages/flutter/lib/src/widgets/scroll_simulation.dart

```diff
@@ -165,67 +166,194 @@ class ClampingScrollSimulation extends Simulation {
-
-    // See getSplineDeceleration().
-    final double deceleration = math.log(0.35 * velocity.abs() / scaledFriction);
-
-    return math.exp(deceleration / (_kDecelerationRate - 1.0));
+  // See getSplineDeceleration().
+  double _splineDeceleration(double velocity) {
+    return math.log(_inflexion *
+        velocity.abs() /
+        (friction * _decelerationForFriction(0.84)));
   }
 
-  // Based on a cubic curve fit to the Scroller.computeScrollOffset() values
-  // produced for an initial velocity of 4000. The value of Scroller.getDuration()
-  // and Scroller.getFinalY() were 686ms and 961 pixels respectively.
-  //
-  // Algebra courtesy of Wolfram Alpha.
-  //
-  // f(x) = scrollOffset, x is time in milliseconds
-  // f(x) = 3.60882×10^-6 x^3 - 0.00668009 x^2 + 4.29427 x - 3.15307
-  // f(x) = 3.60882×10^-6 x^3 - 0.00668009 x^2 + 4.29427 x, so f(0) is 0
-  // f(686ms) = 961 pixels
-  // Scale to f(0 <= t <= 1.0), x = t * 686
-  // f(t) = 1165.03 t^3 - 3143.62 t^2 + 2945.87 t
-  // Scale f(t) so that 0.0 <= f(t) <= 1.0
-  // f(t) = (1165.03 t^3 - 3143.62 t^2 + 2945.87 t) / 961.0
-  //      = 1.2 t^3 - 3.27 t^2 + 3.065 t
-  static const double _initialVelocityPenetration = 3.065;
-  static double _flingDistancePenetration(double t) {
-    return (1.2 * t * t * t) - (3.27 * t * t) + (_initialVelocityPenetration * t);
+  // See getSplineFlingDuration().
+  int _splineFlingDuration(double velocity) {
+    final double deceleration = _splineDeceleration(velocity);
+    return (1000 * math.exp(deceleration / (_kDecelerationRate - 1.0))).round();
   }
 
-  // The derivative of the _flingDistancePenetration() function.
-  static double _flingVelocityPenetration(double t) {
-    return (3.6 * t * t) - (6.54 * t) + _initialVelocityPenetration;
+  // See getSplineFlingDistance().
+  double _splineFlingDistance(double velocity) {
+    final double l = _splineDeceleration(velocity);
+    final double decelMinusOne = _kDecelerationRate - 1.0;
+    return friction *
+        _decelerationForFriction(0.84) *
+        math.exp(_kDecelerationRate / decelMinusOne * l);
   }
 
   @override
   double x(double time) {
-    final double t = (time / _duration).clamp(0.0, 1.0);
-    return position + _distance * _flingDistancePenetration(t) * velocity.sign;
+    if (time == 0) {
+      return position;
+    }
+    final _NBSample sample = _NBSample(time, _duration);
+    return position + (sample.distanceCoef * _distance) * velocity.sign;
   }
 
   @override
   double dx(double time) {
-    final double t = (time / _duration).clamp(0.0, 1.0);
-    return _distance * _flingVelocityPenetration(t) * velocity.sign / _duration;
+    if (time == 0) {
+      return velocity;
+    }
+    final _NBSample sample = _NBSample(time, _duration);
+    return sample.velocityCoef * _distance / _duration * velocity.sign * 1000.0;
   }
 
   @override
   bool isDone(double time) {
-    return time >= _duration;
+    return time * 1000.0 >= _duration;
+  }
+}
+
+class _NBSample {
```

Comment:
To be honest I'm not sure. I think it might just happen to be the variable name of choice for the underlying math equation that the original developer likely referenced. I kept it here as I wanted it to be immediately obvious how this logic is translated from `Scroller.java`.